src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs

```diff
@@ -670,6 +670,136 @@ public void MethodParametersTest(string methodName, NullabilityState stringState
+            for (var i = 0; i < genericArgumentStates.Length; ++i)
+            {
+                if (genericArgumentStates[i] is null)
+                {
+                    Assert.Equal(i, info.GenericTypeArguments.Length);
+                    break;
+                }
+
+                Assert.Equal(genericArgumentStates[i], info.GenericTypeArguments[i].ReadState);
+                Assert.Equal(genericArgumentStates[i], info.GenericTypeArguments[i].WriteState);
+            }
+        }
+
+        [Fact]
+        [SkipOnMono("Nullability attributes trimmed on Mono")]
+        public void ConstrainedGenericMethodWithByRefTest()
+        {
+            // ref T ConstrainedGenericMethodWithByRef<T>(out T[] array) where T : class?
+            MethodInfo method = typeof(TypeWithNotNullContext).GetMethod(nameof(TypeWithNotNullContext.ConstrainedGenericMethodWithByRef))!
+                .MakeGenericMethod(typeof(string));
+
+            NullabilityInfo info = nullabilityContext.Create(method.ReturnParameter);
+            Assert.Equal(method.ReturnType, info.Type);
+            Assert.Equal(NullabilityState.Nullable, info.ReadState);
+            Assert.Equal(NullabilityState.Nullable, info.WriteState);
+
+            info = nullabilityContext.Create(method.GetParameters()[0]);
+            Assert.Equal(typeof(string[]).MakeByRefType(), info.Type);
+            Assert.Equal(NullabilityState.NotNull, info.ReadState);
+            Assert.Equal(NullabilityState.NotNull, info.WriteState);
+            Assert.Equal(typeof(string), info.ElementType!.Type);
+            Assert.Equal(NullabilityState.Nullable, info.ElementType.ReadState);
+            Assert.Equal(NullabilityState.Nullable, info.ElementType.WriteState);
+        }
+
+        [Fact]
+        [SkipOnMono("Nullability attributes trimmed on Mono")]
+        public void MethodWithPointersTest()
+        {
+            // MethodWithPointers(int* a, int?* b)
+            MethodInfo method = typeof(TypeWithNotNullContext).GetMethod(nameof(TypeWithNotNullContext.MethodWithPointers))!;
+            ParameterInfo[] parameters = method.GetParameters();
+
+            NullabilityInfo info = nullabilityContext.Create(parameters[0]);
+            Assert.Equal(parameters[0].ParameterType, info.Type);
+            Assert.Equal(NullabilityState.NotNull, info.ReadState);
+            Assert.Equal(NullabilityState.NotNull, info.WriteState);
+
+            info = nullabilityContext.Create(parameters[1]);
+            Assert.Equal(parameters[1].ParameterType, info.Type);
+            Assert.Equal(NullabilityState.Nullable, info.ReadState);
+            Assert.Equal(NullabilityState.Nullable, info.WriteState);
+        }
+
+        [Fact]
+        [SkipOnMono("Nullability attributes trimmed on Mono")]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/73811", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```

Comment:
Valuetypes require runtime codegen and we didn't generate this code ahead of time.

Can you just add a dummy call to the method being MakeGenericMethod-ed: `if (string.Empty.Length > 0) TypeWithNotNullContext.GenericMethodWithPointers<float>(null, null);`?

Reply:
OK